subworkflows/local/consensus/main.nf

```diff
@@ -1,73 +1,81 @@
+
         ch_versions       = Channel.empty()
-        ch_duplex_metrics = Channel.empty()
-
-        ch_fastq
-          .combine(ch_umi_in_readname)
-          .branch(
-            rn:  { it[3] == true },
-            seq: { it[3] == false }
-          )
-          .set { ch_fastq_branch }
-
-        RN_FQ  = ch_fastq_branch.rn .map { meta, r1, r2, f -> [meta, r1, r2] }
-        SEQ_FQ = ch_fastq_branch.seq.map { meta, r1, r2, f -> [meta, r1, r2] }
-
-        FGBIO_FASTQTOBAM(RN_FQ, ch_reference)
-        ch_versions = ch_versions.mix(FGBIO_FASTQTOBAM.out.versions.first())
-
-        FGBIO_COPYUMIFROMREADNAME(FGBIO_FASTQTOBAM.out.bam)
-        ch_versions = ch_versions.mix(FGBIO_COPYUMIFROMREADNAME.out.versions.first())
-
-        FGBIO_SORTBAM(FGBIO_COPYUMIFROMREADNAME.out.bam)
-        ch_versions = ch_versions.mix(FGBIO_SORTBAM.out.versions.first())
-
-        FGBIO_GROUPREADSBYUMI(FGBIO_SORTBAM.out.bam)
-        ch_versions = ch_versions.mix(FGBIO_GROUPREADSBYUMI.out.versions.first())
-
-        FGBIO_CALLMOLECULARCONSENSUSREADS(FGBIO_GROUPREADSBYUMI.out.bam)
-        ch_versions = ch_versions.mix(FGBIO_CALLMOLECULARCONSENSUSREADS.out.versions.first())
-
-        if (params.enable_duplex) {
-            FGBIO_ZIPPERBAMS(FGBIO_CALLMOLECULARCONSENSUSREADS.out.bam)
-            ch_versions = ch_versions.mix(FGBIO_ZIPPERBAMS.out.versions.first())
-            FGBIO_CALLDUPLEXCONSENSUSREADS(FGBIO_ZIPPERBAMS.out.bam)
-            ch_versions = ch_versions.mix(FGBIO_CALLDUPLEXCONSENSUSREADS.out.versions.first())
-            FGBIO_FILTERCONSENSUSREADS(FGBIO_CALLDUPLEXCONSENSUSREADS.out.bam)
-            ch_versions = ch_versions.mix(FGBIO_FILTERCONSENSUSREADS.out.versions.first())
-            FGBIO_COLLECTDUPLEXSEQMETRICS(FGBIO_FILTERCONSENSUSREADS.out.bam)
-            ch_versions = ch_versions.mix(FGBIO_COLLECTDUPLEXSEQMETRICS.out.versions.first())
-            ch_duplex_metrics = ch_duplex_metrics.mix(FGBIO_COLLECTDUPLEXSEQMETRICS.out.metrics)
-        } else {
-            FGBIO_FILTERCONSENSUSREADS(FGBIO_CALLMOLECULARCONSENSUSREADS.out.bam)
-            ch_versions = ch_versions.mix(FGBIO_FILTERCONSENSUSREADS.out.versions.first())
-        }
-
-        ch_consensus_rn  = FGBIO_FILTERCONSENSUSREADS.out.bam
-        ch_consensus_seq = Channel.empty()   // pending: ExtractUmisFromBam branch
-
-        ch_consensus = ch_consensus_rn.mix(ch_consensus_seq)
+
+        ch_input_fastq
+            .combine(ch_umi_in_readname)
+            .map { meta, fq1, fq2, umi_flag ->
+                tuple(meta, fq1, fq2, umi_flag)
+            }
+            .branch { _meta, _fq1, _fq2, umi_flag ->
+                umi_in_readname: umi_flag == true
+                umi_in_seq: umi_flag == false
+            }
+            .set  {ch_input_fastq_branch}
+
+
+    // Readname branch
+
+        // 1.1: FASTQ => uBAM
+        ch_RN_uBAM = Channel.empty()
+        RN_FQ  = ch_input_fastq_branch.umi_in_readname.map { meta, r1, r2, _f -> tuple(meta, [r1, r2]) }
```

Comment:
A small nitpick but can you make these variable names a bit more clear? (using abbreviations should be avoided unless the names get really long)

Reply:
It's also a good practice to put `def` in front of new variables to declare them as local variables